diff -ur temp/node_modules/daemonize2/lib/daemonize.js node_modules/daemonize2/lib/daemonize.js
--- temp/node_modules/daemonize2/lib/daemonize.js	2016-08-25 11:50:37.793917000 +0200
+++ node_modules/daemonize2/lib/daemonize.js		2016-08-30 19:27:22.942518000 +0200
@@ -68,7 +68,6 @@
         this._options.umask = parseInt(options.umask);
 
     this._options.args = this._makeArray(options.args);
+    this._options.args=["--expose-gc"].concat(this._options.args);
     this._options.argv = this._makeArray(options.argv || process.argv.slice(2));
 
     this._stopTimeout = options.stopTimeout || 2000;
